plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -422,6 +422,10 @@ private static void appendValues(List<Object> list, Object value) {
         throw new IllegalArgumentException("field [" + path + "] of type [" + object.getClass().getName() + "] cannot be cast to [" + clazz.getName() + "]");
     }
 
+    public String renderTemplate(TemplateService.Template template) {
```

Comment:
looks like it only makes sense to re-use in `removeField`. the others do the `copyAndResolve`, which requires `model` to be passed